File: annotationprocessor/src/main/java/org/acra/processor/util/IsValidResourceVisitor.java
package org.acra.processor.util;

import javax.lang.model.util.SimpleAnnotationValueVisitor8;

public class IsValidResourceVisitor extends SimpleAnnotationValueVisitor8<Boolean, Void> {
    @Override
    protected Boolean defaultAction(Object o, Void aVoid) {
        return false;
    }

    @Override
    public Boolean visitInt(int i, Void aVoid) {
        return i != 0;
    }
}

File: annotationprocessor/src/main/java/org/acra/processor/util/IsValidResourceVisitor.kt
package org.acra.processor.util

import javax.lang.model.util.SimpleAnnotationValueVisitor8

class IsValidResourceVisitor : SimpleAnnotationValueVisitor8<Boolean, Unit?>() {
    override fun defaultAction(o: Any, u: Unit?): Boolean {
        return false
    }

    override fun visitInt(i: Int, u: Unit?): Boolean {
        return i != 0
    }
}